packages/runtime/src/provider/provider-server.tsx

```diff
@@ -1,7 +1,8 @@
 import React, { useRef, useContext, createContext } from 'react';
+import { isNodeEnvironment } from '../is-node';
 import { ProviderComponent, UseCacheHook } from './types';
 
-if (process.env.NODE_ENV === 'development' && typeof window !== 'undefined') {
+if (process.env.NODE_ENV === 'development' && isNodeEnvironment()) {
```

Comment:
you have these back to front

if this code runs on the browser - it should throw because it should only run on the server

vice versa for the other.

can you add some tests so we don't accidentally regress this in the future please? 🙏